share/goa/depot/mk/gpg.inc

```diff
@@ -17,9 +17,9 @@ pubkey_path = $(firstword \
 # obtain key ID of 'depot/<user>/pubkey' to be used to select signing key
 pubkey_id = $(shell pubkey_file=$(call pubkey_path,$1); \
                     $(GPG) --yes -o $$pubkey_file.dearmored --dearmor $$pubkey_file; \
-                    $(GPG) --with-colon --no-default-keyring --list-public-keys \
-                           --keyring $$pubkey_file.dearmored |\
-                    head -n 2 | tail -n 1 | cut -d: -f5; rm -f $$pubkey_file.dearmored)
+                    $(GPG) --with-colon --import-options show-only \
+                           --import $$pubkey_file.dearmored |\
```

Comment:
`--import` looks like a capable alternative, especially because it supports armored key files.

Reply:
Thanks. I haven't noticed this in the man, so I can remove dearmoring then and use pubkey file directly. I will update the script.

Reply:
May you please also try to add `--dry-run` to ensure the user keyring is not modified by `--import`? IMO `--import-options show-only` is misleading as it in fact *does change* the keyring. (from the man page)

```
import-show
show-only
    Show a listing of the key as imported right before it is stored.  This can
    becombined with the option --dry-run to only look at keys; the option
    show-only is a shortcut for this combination.  Note that suffixes like '#'
    for "sec" and "sbb" lines  may  or  may  not  be printed.
```

`import-show` is the more accurate alias.